### ISSUE TITLE
no media server with empty media server array in loadBalancer

### ISSUE THREAD

**Issue by shaibiii9777** [davehorton/drachtio-fn-fsmrf-sugar#1]
Hi, 
I am using "callforamerica/freeswitch" docker image to run freeSwitch server, I successfully connect with freeSwitch but always get empty array of mediaservers (to get leastLoadedServer)
following is the loadBalancer object derived from "drachtio-fn-fsmrf-sugar"

LoadBalancer {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  active: [],
  inactive: [ { address: '172.17.0.5', port: 8021, secret: 'ClueCon' } ],
  servers: [ { address: '172.17.0.5', port: 8021, secret: 'ClueCon' } ],
  mrf:
   Mrf {
     _events: [Object: null prototype] {},
     _eventsCount: 0,
     _maxListeners: undefined,
     _srf:
      Srf {
        _events: [Object],
        _eventsCount: 2,
        _maxListeners: undefined,
        _dialogs: Map {},
        _tags: [],
        _app: [Function] },
     debugDir: undefined,
     debugSendonly: undefined,
     mediaservers: [],
     localAddresses:
      [ '10.0.75.1', '192.168.56.1', '192.168.100.67', '172.18.6.49' ],
     customEvents: [] },
  logger:
   { info: [Function: noop],
     debug: [Function: noop],
     error: [Function: noop] },
  timerId:
   Timeout {
     _called: false,
     _idleTimeout: 5000,
     _idlePrev: [TimersList],
     _idleNext: [TimersList],
     _idleStart: 2028,
     _onTimeout: [AsyncFunction: bound _init],
     _timerArgs: undefined,
     _repeat: null,
     _destroyed: false,
     [Symbol(unrefed)]: false,
     [Symbol(asyncId)]: 27,
     [Symbol(triggerId)]: 0 } }

**Comment by davehorton** [comment]
yes, you can't just connect to any freeswitch, you need one configured with a specific dialplan and sip profile.

try `drachtio/drachtio-freeswitch-mrf:1.10.1-v0.2.5`

**Comment by shaibiii9777** [comment]
> 
> 
> yes, you can't just connect to any freeswitch, you need one configured with a specific dialplan and sip profile.
> 
> try `drachtio/drachtio-freeswitch-mrf:1.10.1-v0.2.5`

tried this image also, but this container dosent open any port hence fails connectivity with node
following is the Network setting object from docker inspect container

"NetworkSettings": {
            "Bridge": "",
            "SandboxID": "5b952cd484a86bcdeab21196a1568d900e097a79a51f2dd80d7760156bb956d3",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/5b952cd484a8",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "692f94741a20c909cff9408a02a29b4ab0759e456f6483134efe2a44acdfbc5a",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.5",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:05",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "db8eb4d416e463048b59e37f861077c0c9df37e8b312abd24c1843b1ca0a48aa",
                    "EndpointID": "692f94741a20c909cff9408a02a29b4ab0759e456f6483134efe2a44acdfbc5a",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.5",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:05",
                    "DriverOpts": null
                }
            }
        }

**Comment by shaibiii9777** [comment]
Sorry, My bad, passing logger to the loadBalancer was creating an issue, removed logger and it is working fine, getting mediaserver also, but somehow, application still throwing an error, will get to you again after debugging, this issue can be closed now
